adal/src/telemetry/java/com/microsoft/aad/adal/DefaultEvent.java

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.provider.Settings.Secure;
+import android.text.TextUtils;
+import android.util.Pair;
+
+import java.io.UnsupportedEncodingException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+
+class DefaultEvent implements IEvents {
+    private final List<Pair<String, String>> mEventList;
+
+    private static String sApplicationName = null;
+
+    private static String sApplicationVersion = "NA";
+
+    private static String sClientId = "NA";
+
+    private static String sClientIp = "NA";
+
+    private static String sDeviceId = "NA";
+
+    private String mRequestId;
+
+    private int mDefaultEventCount;
+
+    DefaultEvent() {
+        mEventList = new ArrayList<>();
+
+        // Keying off Application name not being null to decide if the defaults have been set
+        if (sApplicationName != null) {
+            setEvent(EventStrings.APPLICATION_NAME, sApplicationName);
+            setEvent(EventStrings.APPLICATION_VERSION, sApplicationVersion);
+            setEvent(EventStrings.CLIENT_ID, sClientId);
+            setEvent(EventStrings.CLIENT_IP, sClientIp);
+            setEvent(EventStrings.DEVICE_ID, sDeviceId);
+            mDefaultEventCount = mEventList.size();
+        }
+    }
+
+    @Override
+    public int getDefaultEventCount() {
+        return mDefaultEventCount;
+    }
+
+    @Override
+    public void setEvent(final String name, final String value) {
```

Comment:
That name is confusing. it should be `setProperty` - this is normal terms used by the telemetry systems.